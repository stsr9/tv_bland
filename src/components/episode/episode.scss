.episode {
  gap: 10px;
  .image {
    cursor: pointer;
    img {
      @media (max-width: 800px) {
       width: 95%;
      }
    }
  }
  .rating {
    color: orange;
    padding-right: 5px;
    @media (max-width: 800px) {
       display: none;
    }
  }
  .title {
    text-align: left;
    cursor: pointer;
  }
}
