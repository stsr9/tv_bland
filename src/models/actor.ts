export interface Actor {
	person: {
		id: number
		url: string
		name: string
		country: {
			name: string
			code: string
			timezone: string
		}
		birthday: string
		deathday: string
		gender: string
		image: {
			medium: string
			original: string
		}
		_links: {
			self: {
				href: string
			}
		}
	}
	character: {
		id: number
		url: string
		name: string
		image: string
		_links: {
			self: {
				href: string
			}
		}
	}
	self: boolean
	voice: boolean
}
