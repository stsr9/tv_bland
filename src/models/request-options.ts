type HttpMethod = 'get' | 'post' | 'put' | 'delete' | 'patch'

export type QueryParams = {
	[key: string]: string | number | boolean | undefined | null
}

export interface RequestOptions {
	headers?: {
		[key: string]: string
	}
	queryParams?: QueryParams
}

export interface ExtendedRequestOptions<T> extends RequestOptions {
	method: HttpMethod
	url: string
	body?: T
}
