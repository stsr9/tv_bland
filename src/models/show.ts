export interface Show {
    id: number
    url: string
    name: string
    type: string
    language: string
    genres: string[]
    status: string
    runtime: number
    averageRuntime: number
    premiered: string
    officialSite: string
    schedule: {
        time: string
        days: string[]
    }
    rating: {
        average: number
    }
    weight: number
    network: {
        id: number
        name: string
        country: {
            name: string
            code: string
            timezone: string
        }
    }
    webChannel: {
        id: number
        name: string
        country: {
            code: string
            name: string
            timezone: string
        }
    }
    dvdCountry: string | null
    externals: {
        tvrage: number
        thetvdb: number
        imdb: string
    }
    image: {
        medium: string
        original: string
    }
    summary: string
    updated: number
    _links: {
        self: {
            href: string
        }
        previousepisode: {
            href: string
        }
        nextepisode: {
            href: string
        }
    }
}
