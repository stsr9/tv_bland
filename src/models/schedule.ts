import { Show } from './show'

export interface Schedule {
	id: number
	url: string
	name: string
	season: number
	number: number
	type: string
	airdate: string
	airtime: string
	airstamp: string
	runtime: number
	image: string | null
	summary: string | null
	show: Show
	_links: {
		self: {
			href: string
		}
	}
}
