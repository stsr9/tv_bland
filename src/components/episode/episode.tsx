import { Schedule } from 'models/schedule'
import './episode.scss'
import Rating from 'react-rating'
import { Link } from 'react-router-dom'
import { calcRating } from 'utils/rating'

type Props = {
	episode: Schedule
}

export const Episode = ({ episode }: Props) => {
	return (
		<div className="episode flex-column">
			<div className="image">
				<Link to={`/episodes/${episode.show.id}`}>
					<img alt={episode.name} src={episode.show.image?.medium} />
				</Link>
			</div>
			<div className="rating">
				<Rating
					readonly
					initialRating={calcRating(episode.show?.rating.average)}
					emptySymbol="fa fa-star-o medium"
					fullSymbol="fa fa-star medium"
				/>
			</div>
			<div className="title">{episode.show?.name}</div>
		</div>
	)
}
