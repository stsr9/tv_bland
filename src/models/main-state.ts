import { Schedule } from './schedule'
import { Show } from './show'
import { Actor } from './actor'

export interface MainState {
	episodesState: {
		episodes: Schedule[]
		episodeDetails: Show
	}
	castState: {
		cast: Actor[]
	}
}
