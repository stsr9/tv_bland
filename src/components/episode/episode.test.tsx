import renderer from 'react-test-renderer'
import { Episode } from './episode'
import { Schedule } from 'models/schedule'
import { BrowserRouter as Router } from 'react-router-dom'

const episode = {
    show: {
        id: 123,
        name: 'best show ever',
        url: 'http://image-url',
        rating: {
            average: 7.6
        }
    }
}

it('renders without crashing', () => {
  const episodeDetails = renderer.create(
      <Router>
          <Episode
            episode={episode as Schedule}
          />
      </Router>
  ).toJSON();

  expect(episodeDetails).toMatchSnapshot();
});
