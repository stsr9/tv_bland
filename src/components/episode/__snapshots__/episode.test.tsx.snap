// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="episode flex-column"
>
  <div
    className="image"
  >
    <a
      href="/episodes/123"
      onClick={[Function]}
    >
      <img />
    </a>
  </div>
  <div
    className="rating"
  >
    <span
      style={
        Object {
          "direction": "ltr",
          "display": "inline-block",
        }
      }
    >
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "inherit",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "visibility": "hidden",
            }
          }
        >
          <span
            className="fa fa-star-o medium"
          />
        </span>
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        >
          <span
            className="fa fa-star medium"
          />
        </span>
      </span>
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "inherit",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "visibility": "hidden",
            }
          }
        >
          <span
            className="fa fa-star-o medium"
          />
        </span>
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        >
          <span
            className="fa fa-star medium"
          />
        </span>
      </span>
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "inherit",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <span
          style={
            Object {
              "visibility": "hidden",
            }
          }
        >
          <span
            className="fa fa-star-o medium"
          />
        </span>
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "100%",
            }
          }
        >
          <span
            className="fa fa-star medium"
          />
        </span>
      </span>
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "inherit",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <span
          style={Object {}}
        >
          <span
            className="fa fa-star-o medium"
          />
        </span>
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "79.99999999999999%",
            }
          }
        >
          <span
            className="fa fa-star medium"
          />
        </span>
      </span>
      <span
        onClick={[Function]}
        onMouseMove={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        style={
          Object {
            "cursor": "inherit",
            "display": "inline-block",
            "position": "relative",
          }
        }
      >
        <span
          style={Object {}}
        >
          <span
            className="fa fa-star-o medium"
          />
        </span>
        <span
          style={
            Object {
              "display": "inline-block",
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": 0,
              "width": "0%",
            }
          }
        >
          <span
            className="fa fa-star medium"
          />
        </span>
      </span>
    </span>
  </div>
  <div
    className="title"
  >
    best show ever
  </div>
</div>
`;
